src/codegen/compile.ml

```diff
@@ -2470,15 +2769,16 @@ module TaggedSmallWord = struct
     in lsb_adjust ty ^^ set_b ^^ lsb_adjust ty ^^
        compile_unboxed_one ^^ get_b ^^ clamp_shift_amount ty ^^
        G.i (Binary (Wasm.Values.I32 I32Op.Shl)) ^^
-       G.i (Binary (Wasm.Values.I32 I32Op.And))
+       G.i (Binary (Wasm.Values.I32 I32Op.And)) ^^
+       msb_adjust ty
```

Comment:
I am not sure whether this `msb_adjust ty` is really needed because the result should be a boolean (not `ty`).

Reply:
prim.mo expects it to return a ty, which is then equated against 0:ty. I think there's an existing todo to fix this....
I think I'll leave it for now...

Reply:
Created an issue to fix this https://github.com/dfinity/motoko/issues/4391